src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -293,6 +298,45 @@ private void AssertVisited(BoundExpression expr)
             throw ExceptionUtilities.Unreachable();
         }
 
+        // Unlike the base implementation, we want to visit all the nodes, not just the operands.
+        public override BoundNode? VisitBinaryOperator(BoundBinaryOperator node)
```

Comment:
>VisitBinaryOperator [](http://example.com/codeflow?start=35&length=19)

I do not see any good reason to use drastically different algorithm in this method by comparison to ```BoundTreeWalkerWithStackGuardWithoutRecursionOnTheLeftOfBinaryOperator.VisitBinaryOperator``` implementation. This could be explained by desire to call ```VisitCore``` on the way "out" rather than on the way "in", but I think we actually want to do that the other way.